.ci/embedded-64bit-steps.yaml

```diff
@@ -0,0 +1,139 @@
+      sudo tar -C /opt/ -xvpf julia-1.3.0-linux-x86_64.tar.gz
+    fi
+
+    echo "Download and build boost as a static lbrary"
+        wget --no-check-certificate \
+        "https://dl.bintray.com/boostorg/release/1.75.0/source/boost_1_75_0.tar.gz" && \
+        tar xvzf boost_1_75_0.tar.gz && \
+        rm -f boost_1_75_0.tar.gz && \
+
+    echo "Download and build OpenBLAS as a static library"
+    wget --no-check-certificate \
+      "https://github.com/xianyi/OpenBLAS/archive/v0.3.13.tar.gz" && \
+       tar xvzf v0.3.13.tar.gz && \
+       rm -f v0.3.13.tar.gz && \
+       cd OpenBLAS-0.3.13 && \
+       make TARGET=ARMV8 BINARY=64 HOSTCC=gcc CC=aarch64-linux-gnu-gcc FC=aarch64-linux-gnu-gfortran NO_SHARED=1 &&\
+       cd ../ &&\
+ 
+    # Install armadillo.
+    curl https://data.kurg.org/armadillo-8.400.0.tar.xz | tar -xvJ
+
+    # Install cereal.
+    wget https://github.com/USCiLab/cereal/archive/v1.3.0.tar.gz
+    tar -xvzpf v1.3.0.tar.gz # Unpack into cereal-1.3.0/.
+    ls -la
+    pwd
+  displayName: 'Install Build Dependencies'
+
+# Configure mlpack (CMake)
+- script: |
+    unset BOOST_ROOT
+    mkdir build && cd build
+    if [ "$(binding)" == "go" ]; then
+      export GOPATH=$PWD/src/mlpack/bindings/go
+      go get -u -t gonum.org/v1/gonum/...
+    fi
+    cmake $(CMakeArgs) -DPYTHON_EXECUTABLE=`which python` ..
+  displayName: 'CMake'
+
+# Build mlpack
+- script: cd build && make
+  condition: eq(variables['CMakeArgs'], '-DDEBUG=ON -DPROFILE=OFF -DBUILD_PYTHON_BINDINGS=OFF -DBUILD_JULIA_BINDINGS=OFF -DBUILD_GO_BINDINGS=OFF -DBUILD_R_BINDINGS=OFF')
+  displayName: 'Build'
+
+# Build mlpack
+- script: cd build && make -j2
+  condition: ne(variables['CMakeArgs'], '-DDEBUG=ON -DPROFILE=OFF -DBUILD_PYTHON_BINDINGS=OFF -DBUILD_JULIA_BINDINGS=OFF -DBUILD_GO_BINDINGS=OFF -DBUILD_R_BINDINGS=OFF')
+  displayName: 'Build'
+
+- script: |
+    uname -m
+    docker pull arm64v8/ubuntu
+    docker run --rm -t arm64v8/ubuntu uname -m
+    docker pull multiarch/qemu-user-static
+    docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+    tar -cvzf mlpack.tar.gz /home/vsts/work/1/s
+    mv mlpack.tar.gz /home/vsts/work/1/s/.ci/aarch64
+  displayName: 'Download docker container'
+
+# #Allow running docker container with different architecture than the host
+# - task: Docker@2
+#   displayName: 'Run privileged container'
+#   inputs:
+#     command: run
+#     arguments: '--rm --privileged multiarch/qemu-user-static:register --reset -p yes'
+#   enabled: false
+
+# Start docker container
+# - task: Docker@2
+#   displayName: 'Run tests via ctest'
+#   inputs:
+#     command: run
+#     arguments: '-v /home/vsts/work/1/s:/opt/share -t arm64v8/ubuntu'
+#     tags: arm64v8/ubuntu
+#     script: | # maybe the name is different?
+#         uname -m
+#         ls
+#         whoami
+#   #addPipelineData: false
+
+# Start docker container
+- task: Docker@2
```

Comment:
I realize that you've spent a *lot* of time and attempts fighting with this, so I might be jumping in with ideas you tried many days ago... :)

I was taking a look through the Azure CI documentation, and I think maybe you don't need a `Docker@2` task.  To my understanding, that is a task that can build a container, but technically you don't need to build a container---you only need to run the tests inside of a container.  I can see above (lines 77-85) that you've got a script block that pulls the `arm64v8/ubuntu` container and then builds a new container with that as a base.

I think maybe it could be simpler if instead of building a Docker image, you ran the `arm64v8/ubuntu` image and just had a script that you ran inside of it.  So imagine having a script file like this:

```sh
# This script installs all necessary packages and then runs the tests.
apt-get update
apt-get install -y --allow-unauthenticated cmake
cd /build/
# just a debugging pwd here... I *think* this will be in the root of the build directory, but good to check
pwd
# same, check if we are in the build directory
ls

# Now actually run the tests.
CTEST_OUTPUT_ON_FAILURE=1 ctest -T Test .
```

And then in this yaml file, have a script step like this:

```
  docker pull arm64v8/ubuntu
  docker run -v .:/build/ --rm arm64v8/ubuntu test_script.sh
```

Anyway, I am not totally sure if that will work, but at least to me it seems like we should be able to avoid building a new Docker image just to run the tests.  I think we can just use the existing `arm64v8/ubuntu` one.

Reply:
I will give it a try, thanks